web/src/bootstrap_typeahead.ts

```diff
@@ -361,10 +380,6 @@ class Typeahead<ItemType extends string | object> {
         return this.render(final_items.slice(0, this.items), matching_items).show();
     }
 
-    defaultMatcher(item: string): boolean {
-        return item.toLowerCase().includes(this.query.toLowerCase());
-    }
-
     defaultSorter(items: string[]): string[] {
```

Comment:
Is this copy supposed to be removed in this commit? I see it appearing above.

Reply:
It's not a copy, I just renamed it here for clarity. A future commit (that should eventually be squashed if it's doing the right thing) involves moving this function out of the class into a standalone function.

Reply:
Read https://github.com/zulip/zulip/pull/29266/commits/c475363487356ebda5ceb9ce2ce97f7556171a8f#diff-0e7057e3f4e0274939027a49805e321b5d97540e4c4d378ec7e9d86ef346d36c. You add a copy of `defaultSorter` above, but don't delete the one here, below.

Reply:
I'm not sure this approach is better than just having these functions declared as accepting `ItemType` and just asserting in their bodies that the thing passed is a string before doing string operations on them.

Certainly `defaultUpdater` would be cleaner done that way -- just type it like this:
```
    defaultUpdater(item: ItemType): ItemType {
        return item;
    }
```

Reply:
> but don't delete the one here, below.

Oh yeah, my mistake ,thanks.

> asserting in their bodies that the thing passed is a string before doing string operations on them.

This works for matcher and updater, but I can't figure out how to get it working for `sorter` because it doesn't like that it's returning something of type string. I'll update the other two for now and keep `sorter` as it was with the first solution.